src/Stratis.Bitcoin/Connection/ConnectionManager.cs

```diff
@@ -141,6 +141,22 @@ public void Initialize()
 
             this.StartNodeServer();
 
+            // if external IP address supplied this overrides all.
+            if (this.ConnectionSettings.ExternalEndpoint != null)
+            {
+                this.selfEndpointTracker.MyExternalAddress = this.ConnectionSettings.ExternalEndpoint;
+                this.selfEndpointTracker.IsMyExternalAddressFinal = true;
```

Comment:
This logic is correct, but I would actually use the update method even here instead of this direct modification of properties that belong to different components. You basically just need a new parameter in the update method saying that it is final.